### PR TITLE
B #5325 Fixed snapshot-delete problem.

### PR DESCRIPTION
**Pull request by Semedi**

### CODE REVIEW

File: src/cli/onevm [hunk 1]
```diff
@@ -162,7 +162,9 @@ cmd=CommandParser::CmdParser.new(ARGV) do
     end
 
     set :format, :vmid, OneVMHelper.to_id_desc do |arg|
-        helper.to_id(arg)
+        tmp = helper.to_id(arg)
+        @current_vm=tmp[1]
```
Comment (jfontan) [inline src/cli/onevm]: Spaces before and after equal sign

File: src/cli/one_helper/onevm_helper.rb [hunk 2]
```diff
@@ -192,6 +192,15 @@ def self.ip_str(vm)
         end
     end
 
+    def retrieve_snapshot_id(vm_id, id)
+        name = id
+        vm = retrieve_resource(vm_id)
+        vm.info
+        id = vm.retrieve_elements("/VM/TEMPLATE/SNAPSHOT[NAME='#{name}']/SNAPSHOT_ID") unless /\A\d+\z/.match(name)
+
+        result = id.nil? || id.size > 1 ? [-1, "#{name} not found or duplicated"] :[0, id[0].to_i]
```
Comment (jfontan) [inline src/cli/one_helper/onevm_helper.rb]: `result =` is not needed. Also an `if` construct seems more appropriate here.

File: src/cli/one_helper/onevm_helper.rb [hunk 3]
```diff
@@ -192,6 +192,15 @@ def self.ip_str(vm)
         end
     end
 
+    def retrieve_snapshot_id(vm_id, id)
+        name = id
+        vm = retrieve_resource(vm_id)
+        vm.info
```
Comment (jfontan) [inline src/cli/one_helper/onevm_helper.rb]: Check info response

File: src/cli/one_helper/onevm_helper.rb [hunk 4]
```diff
@@ -192,6 +192,15 @@ def self.ip_str(vm)
         end
     end
 
+    def retrieve_snapshot_id(vm_id, id)
+        name = id
```
Comment (jfontan) [inline src/cli/one_helper/onevm_helper.rb]: Why using an extra `name` variable?

File: src/cli/one_helper/onevm_helper.rb [hunk 5]
```diff
@@ -192,6 +192,22 @@ def self.ip_str(vm)
         end
     end
 
+    def retrieve_snapshot_id(vm_id, id)
+
+        vm = retrieve_resource(vm_id)
+        vm.info
```
Comment (Semedi) [inline src/cli/one_helper/onevm_helper.rb]: mandatory

**Pull request merged**